lib/datadog/appsec/instrumentation/gateway/object.rb

```diff
@@ -0,0 +1,24 @@
+# typed: ignore
+# frozen_string_literal: true
+
+
+# typed: false
+# frozen_string_literal: true
+
+module Datadog
+  module AppSec
+    module Instrumentation
+      class Gateway
+        class Object; end
+
+        class User < Object
+          attr_reader :id
+
+          def initialize(id)
```

Comment:
I somewhat shudder at re-using the word `Object`, since it's something really core to Ruby as well. Of course this doesn't clash for the programming language itself but... it may be really confusing from time to time.

Maybe consider `Payload`? `Argument`? `Value`? Something like that?

Reply:
Absolutely. We are experimenting with the naming and implementation. Still, totally agree that the object might be too generic and also lead to confusion with the ruby `Object` class. 

I like `Argument`